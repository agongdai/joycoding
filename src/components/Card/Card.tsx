import React from 'react';

export default function Card() {
  return (
    <div>
      <h4>Card Header</h4>
      <div>Card Body</div>
    </div>
  );
}
