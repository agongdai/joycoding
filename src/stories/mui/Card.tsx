import React from 'react';

import Box from '@mui/material/Box';
import Popover from '@mui/material/Popover';
import Typography from '@mui/material/Typography';

export interface CardProps {
  label: string;
  value: string;
  open: boolean;
}

export const Card = ({ label, value, open }: CardProps): React.ReactElement => {
  const [anchorEl, setAnchorEl] = React.useState<HTMLDivElement | null>(null);

  const handleClick = (event: React.MouseEvent<HTMLDivElement>): void => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = (): void => {
    setAnchorEl(null);
  };

  return (
    <div>
      <Box
        onClick={handleClick}
        sx={{
          display: 'flex',
          justifyContent: 'space-between',
          alignItems: 'center',
          backgroundColor: 'white',
          border: '1px solid #E5E5E5',
          borderRadius: '4px',
          width: '22rem',
          padding: '8px 16px',
        }}
      >
        <div>
          <p style={{ color: '#6C7384' }}>{label}</p>
          <p style={{ fontWeight: 'bold', color: '#1B2026' }}>{value}</p>
        </div>
      </Box>
      <Popover
        open={open}
        anchorEl={anchorEl}
        onClose={handleClose}
        sx={{ top: '150px', left: '28px' }}
      >
        <Typography sx={{ p: 2 }}>The content of the Popover.</Typography>
      </Popover>
    </div>
  );
};

Card.defaultProps = {
  label: 'Heading 5',
  value: 'center',
};
