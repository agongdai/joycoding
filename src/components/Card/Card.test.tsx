import { render, screen } from '@testing-library/react';

import Card from './Card';

describe('Home', () => {
  it('renders a heading', () => {
    const { container } = render(<Card />);

    const heading = screen.getByRole('heading', {
      name: /Card Header/i,
    });

    expect(heading).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
