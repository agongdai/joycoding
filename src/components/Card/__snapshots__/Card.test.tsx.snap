// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <div>
    <h4>
      Card Header
    </h4>
    <div>
      Card Body
    </div>
  </div>
</div>
`;
